flathunter/crawl_idealista.py

```diff
@@ -0,0 +1,68 @@
+"""Expose crawler for Idealista"""
+import logging
+import re
+
+import requests
+from bs4 import BeautifulSoup
+from flathunter.abstract_crawler import Crawler
+from random_user_agent.user_agent import UserAgent
+
+class CrawlIdealista(Crawler):
+    """Implementation of Crawler interface for Idealista"""
+
+    __log__ = logging.getLogger('flathunt')
+    URL_PATTERN = re.compile(r'https://www\.idealista\.it')
+
+    def __init__(self, config):
+        logging.getLogger("requests").setLevel(logging.WARNING)
+
+    # pylint: disable=unused-argument
+    def get_page(self, search_url, driver=None, page_no=None):
+        """Applies a page number to a formatted search URL and fetches the exposes at that page"""
+        return self.get_soup_with_proxy(search_url)
```

Comment:
And test for self.config.use_proxy() here, to decide which get_page function to call.